### DISCUSSION TITLE
How do I migrate an account?

### DISCUSSION THREAD

**Discussion by kytta** [pixelfed/pixelfed#4661] [Q&A] [answered]
I want to migrate a Pixelfed account from one instance (pixelfed.social, for that matter) to a smaller one.

In the account settings, there are ‘Account Aliases’. The help text says: “To move from another account to this one, first you need to create an alias.” I went to the new instance and create an alias with the old account’s username.

Well, what’s next? The Help page on the topic sends me to the ‘Account Aliases’. The ‘Account Aliases’ page says: “The process of moving the account starts from the old one.”

Where is the button/link that initiates the migration?

**Comment by silsha** [comment]
I have the same problem. The description is extremely vague. After reading through the code for an hour now, I'm not even sure if migration between different instances is possible, or only changing the username within the same instance.

**Reply by silsha** [reply]
On the other hand, the help page says: "To migrate your account successfully, your old account must be on a Pixelfed or Mastodon server […]". The wording and the mention of "Mastodon" sounds to me like instance migrations should be possible …

Edit: The help pages also mention "when enabled by server admins." but I can't find such setting anywhere.

Perhaps @dansup has a moment to clarify.

**Reply by silsha** [reply]
Soo ... what I have found out so far:

1. Setting the old username on the new profile seems to correctly set the "alsoKnownAs" value in the ActivityPub user endpoint. This is the first part of being recognised as "moved".
	
	![grafik](https://github.com/pixelfed/pixelfed/assets/31850/95daa934-c4a1-4e1d-9506-37a8b09220e7)

2. The second part of this process would be the reference of the old account to the new profile by means of the "movedTo" key. For this, the code seems to check if there is a profile ID in the `moved_to_profile_id` of the `profiles` table for this user. The profile ID must be another profile in the same table - so your old instance must have retrieved the (new) profile in the past (e.g. by opening the profile page).
3. I manually set this field to the ID of my new profile, and after a while the value of "movedTo" was displayed:
	![grafik](https://github.com/pixelfed/pixelfed/assets/31850/15958b45-7993-4c56-960c-21094def63ad)

Now at least mastodon recognised and displayed my profile as moved:

![grafik](https://github.com/pixelfed/pixelfed/assets/31850/889e1b48-e9f7-45c5-b4a2-982a36dc676b)

What doesn't seem to work?

* It seems that my old instance should send a signed move request to my followers' instances. I think this would make them automatically follow my new profile. That doesn't happen - and I couldn't find any code to do it.
* My posts weren't migrated - again, couldn't find any code or CLI command. (I just re-posted all my photos to the new instance).
* Pixelfed doesn't show my old profile as migrated.

**Comment by meerjungfrauengrotte** [comment]
Same problem here, didn't find a solution. Both Pixelfed instances are up-to-date.

**Comment by ghost** [comment]
Same here - both instances are on latest commits of 0.11.9 and the new instance has been equipped with the account alias of the old instance. Anyway, nothing happens and despite the announcements, this wanted feature is documented really vague - there is no button to trigger it, no option to activate this feature nor any parameter you may enable it with in the .env-file, at least if using the official documentation... Any info from @dansup would be appreciated!

**Comment by rwetzlmayr** [comment]
See also issue #814 opened on on Feb 6, 2019.

**Comment by coastgnu** [comment]
[@dansup@mastodon.social](https://norden.social/@dansup@mastodon.social) 
[@dansup@pixelfed.social](https://norden.social/@dansup@pixelfed.social)
[@pixelfed](https://norden.social/@pixelfed@mastodon.social)
[@coastGNU@pixelfed.social](https://norden.social/@coastGNU@pixelfed.social)

Any ideas?

### Migration from Pixelfed to Mastodon

In Discussions there is an open topic, #4661, ['How.do I migrate an.account?'](https://github.com/pixelfed/pixelfed/discussions/4661) which is still unanswered!

And it seems to me [#migration](https://norden.social/tags/migration) from [#pixelfed](https://norden.social/tags/pixelfed) to [#mastodon](https://norden.social/tags/mastodon) still doesn't work? 
Even when it was announced back in 2023, what went wrong?

### What I did:
 - I added an alias [@coastGNU@pixelfed.social](https://norden.social/@coastGNU@pixelfed.social) in my norden.social account
 - I logged in to my pixelfeed.social account
 - I searched for my norden.social account and opened this profile in pixelfed.social

#### What I expected
 - pixelfed will start migrating my followers and my followed accounts to norden.social

#### What happened
 - nothing, no migration at all

So the question still is:

### How to migrate a pixelfeed account from pixelfeed.social to norden.social?

Any hint or help would be appreciated!

**Comment by forenta** [comment, marked as answer]
https://github.com/pixelfed/pixelfed/pull/4968

**Reply by nigini** [reply]
It looks like this answer is marked as correct but it is not really the case. That pull request only described some background work that was not added to the user interface. For all that matters, there is no clear answer to  "How do I migrate an account?"

Would you agree?

**Reply by kytta** [reply]
I'll answer since I've marked it as correct. Yes, it doesn't give one directions to migrate an account, but in my case, it does answer the question. As in:

— How to migrate accounts?
— You can't (yet), here's the issue

**Reply by nigini** [reply]
Sounds good, @kytta. Thanks for your explanation. I think it will help people that come following this.

**Reply by forenta** [reply]
> It looks like this answer is marked as correct but it is not really the case.

It wasn't even meant to be the _correct_ answer. 🤨

**Reply by ashwinvis** [reply]
Now there is a page to do this. See for example:

https://pixelfed.social/settings/account/migration/manage

EDIT: I tried to move from Pixelfed instance to Pixelfed instance, but I get the same behavior as:

https://github.com/pixelfed/pixelfed/discussions/4661#discussioncomment-8425169

Old account shows a redirected link, but no followers moved.

**Reply by trufae** [reply]
This feature should be priorized.
